src/context/product-listing/product-context.jsx

```diff
@@ -0,0 +1,77 @@
+import { createContext, useContext, useReducer } from 'react';
+
+const initialState = {
+  priceSort: null,
+  inStock: false,
+  fastDelivery: null,
+  priceRange: 5000,
+  productsDispatch: () => {},
+};
+
+const ProductsContext = createContext(initialState);
+const reducer = (state, action) => {
```

Comment:
It would be better if you rename your reducer to productReducer or something of that sort and extract out your reducer in a separate file, the code would look much cleaner.